Fix formatting of currentDay

Abban opened the pull request:
This was being formatted incorrectly for German banners, with the day coming after the month instead of before.

I added the day as a template tag in the messages to fix this.

gbirke commented:
I don't want the code to search for the tracker forever, so I need some kind of limit to the timeout. However, that becomes hard to test, because then the test times out or runs too long, waiting unnecessarily. I've updated the code to implement better intervals (wait for 100, 200, 300 ... ms) and pass in the re-scheduling as a function for better testability. 

This is a very good example of why code should be mostly side-effect free and how to isolate your code for side-effects. The scheduling function itself (timeout + to produce the correct sequence of millisecond intervals) is now untested, but the important part (that it stops after 10 tries) is still tested. I've user-tested it though and it works.

Abban committed:
Fix formatting of currentDay

This was being formatted incorrectly for German banners,
with the day coming after the month instead of before.

I added the day as a template tag in the messages to fix
this.

gbirke reviewed (approved):
Could you make the German ordinal formatter more semantic by adding the dot in the format function and removing it from the messages? You might need to adapt the campaign day sentence as well

Abban committed:
Move dot to DE ordinal formatter

As all German ordinals have a trailing dot this can
be output by the formatter rather than included in
all messages that use one.
